Inventory.md

```diff
@@ -0,0 +1,173 @@
+
+Making sure related functionality (fee-estimation, coin selection) could maybe be subtly difficult. (@paweljakubas?)
+
+### Tx inputs
+
+The same goes for the different transaction inputs:
+- We previously had a `Tx` abstraction for supporting inputs with and without `coin`-values. Removing it was conceptually simple but touched a lot of code. Before re-adding it we could take the opportunity to think of alternative solutions. ⚠️
+- The coins in the Jörmungandr tx inputs are exposed in the Wallet API. Something needs to be changed to be able to serve Haskell Byron tx inputs in the API. If we want to continue exposing them in for the Jörmungandr-compatible wallet we need some added abstraction in the wallet API code. ⚠️
+
+### Addresses
+
+We can trivially re-encode addresses to work with the wallet core and re-decode them later
+(when checking if they are ours or not).
+
+We might be able to treat addresses more abstractly and avoid this.
+
+### Active slot coeff
+
+While missing in Byron, it will exist in Shelley. We could set it to 100% internally in the wallet. Should be a problem at all. ✅
+
+### Superfluous prevBlockHash in BlockHeader
+
+With a good abstraction, or dropped Jörmungandr support we could remove the prevBlockHash field in BlockHeader, but it does no damage there. ✅
+
+### Point ByronBlock vs BlockHeader
+
+We can probably get by with magic values to represent the `Origin`-case. ✅
+
+A way to treat points (like ourobouros-network's`Point ByronBlock`) more abstract in the wallet could potentially be needed to avoid sabotaging other abstractions.
+
+### SlotId vs SlotNo
+
+When slotsPerEpoch cannot change the conversion between SlotId and SlotNo is simple.
+
+But to support protocol parameter updates, we should consider making the core wallet logic deal with SlotNo for the Haskell node.
+
+For Byron, we can assume they are static. ✅
+
+### Epoch Bounday Blocks
+
+I imagine having two blocks with the same slot might break assumptions in our DB-rollbacks, but I don't know.
+
+A simple solution would be to filter them out. ✅
+
+For mainnet Shelley, EBBs do provide an easy way of knowing when a new epoch starts without having to keep track of protocol parameters (and when slots are counted from genesis). This would be an argument for not filtering them out. We will however most likely be maintaining up-to-date protocol parameters anyway, and would be able to tell when we're entering a new epoch through the protocol-params and the current slotNo.
+
+### No rollbacks further than "k"
+
+With the Haskell node we will be able to say that old enough state/checkpoints are immutable and can never be rolled back. We might want to change which checkpoints we store in DB. There is no point in keeping multiple checkpoints in the immutable part of the chain.
+
+### Delegation features
+
+We cannot implement delegation features. We could
```

Comment:
recent Matthias endpoint type magic could help us here probably

Reply:
> type magic refactoring

I think that was the `Api.Link` thing, which I think would only reduce boilerplate in the integration tests. Shouldn't help us here.

Matthias pointed that the wallet already is modular enough though. https://github.com/input-output-hk/cardano-wallet/pull/1214#discussion_r364637707